src/FlatSharp.Compiler/FlatSharpCompiler.cs

```diff
@@ -263,25 +264,53 @@ internal static byte[] GetBfbs(CompilerOptions options)
             Directory.CreateDirectory(outputDir);
             FileInfo info = new FileInfo(options.InputFile);
 
-            System.Diagnostics.Process p = new System.Diagnostics.Process
+            using var p = new Process
             {
-                StartInfo = new System.Diagnostics.ProcessStartInfo
+                StartInfo = new ProcessStartInfo
                 {
                     CreateNoWindow = true,
                     RedirectStandardError = true,
                     RedirectStandardOutput = true,
                     UseShellExecute = false,
                     FileName = flatcPath,
-                    Arguments = $"-b --schema --bfbs-comments --bfbs-builtins --bfbs-filenames \"{info.DirectoryName}\" --no-warnings -o \"{outputDir}\" \"{info.FullName}\"",
                 }
             };
 
+            foreach (var arg in new[]
+            {
+                "-b",
+                "--schema",
+                "--bfbs-comments",
+                "--bfbs-builtins",
+                "--bfbs-filenames",
+                info.DirectoryName,
+                "--no-warnings",
+                "-o",
+                outputDir,
+                info.FullName
+            })
+            {
+                p.StartInfo.ArgumentList.Add(arg);
+            }
+
             try
             {
-                p.Start();
-                string stdout = p.StandardOutput.ReadToEnd();
-                string stderr = p.StandardError.ReadToEnd();
+                p.EnableRaisingEvents = true;
+
+                var lines = new List<string>();
+
+                void OnDataReceived(object sender, DataReceivedEventArgs args)
+                {
+                    if (args.Data is { } line)
+                        lines.Add(line);
+                }
+
+                p.OutputDataReceived += OnDataReceived;
+                p.ErrorDataReceived += OnDataReceived;
 
+                p.Start();
```

Comment:
No there isn't. My understanding is that if the spawned process produces output, it will sit in a buffer until it's read and the child process will block if the buffer becomes full before reading starts and has a chance to drain it (which can be the source of deadlock when reading output and error streams synchronously). That said, [`BeginOutputReadLine`] will anyway [throw if called before `Start`](https://github.com/dotnet/runtime/blob/12a8819eee9865eb38bca6c05fdece1053102854/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs#L1578-L1581) (same for `BeginErrorReadLine`). The [remarks section of the documentation on `BeginOutputReadLine`](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.beginoutputreadline?view=net-6.0#remarks) also says to start the process before beginning to read:

> Follow these steps to perform asynchronous read operations on [`StandardOutput`] for a [`Process`] :
>
> 1. Set [`UseShellExecute`] to `false`.
> 1. Set [`RedirectStandardOutput`] to `true`.
> 1. Add your event handler to the [`OutputDataReceived`] event. The event handler must match the [`System.Diagnostics.DataReceivedEventHandler`] delegate signature.
> 1. Start the [`Process`].
> 1. Call [`BeginOutputReadLine`] for the [`Process`]. This call starts asynchronous read operations on [`StandardOutput`].

[`BeginOutputReadLine`]: https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.beginoutputreadline?view=net-6.0
[`Process`]: https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process?view=net-6.0
[`OutputDataReceived`]: https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.outputdatareceived?view=net-6.0
[`RedirectStandardOutput`]: https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.redirectstandardoutput?view=net-6.0
[`StandardOutput`]: https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.standardoutput?view=net-6.0
[`System.Diagnostics.DataReceivedEventHandler`]: https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.datareceivedeventhandler?view=net-6.0
[`UseShellExecute`]: https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.useshellexecute?view=net-6.0

Reply:
Wow, thanks! I appreciate you being so thorough!